bundles/remote_services/discovery_zeroconf/src/discovery_zeroconf_announcer.c

```diff
@@ -532,7 +644,8 @@ static void *discoveryZeroconfAnnouncer_refreshEndpointThread(void *data) {
                 discoveryZeroconfAnnouncer_handleMDNSEvent(announcer);
             }
         } else if (result == -1 && errno != EINTR) {
-                celix_logHelper_error(announcer->logHelper, "Announcer: Error Selecting event, %d.", errno);
+            celix_logHelper_error(announcer->logHelper, "Announcer: Error Selecting event, %d.", errno);
+            sleep(1);//avoid busy loop
```

Comment:
I'm not familiar with this part yet, and thus will only leave my two cents here.

While `errno == ENOMEM` suggests a temporary failure,  for which `sleep(1)` does make sense, `errno == EBADF` may imply that the bundle could not work any  more, e.g. the `eventFd` or `dsFd` may be accidentally closed by some buggy code.

We can really do nothing locally to recover from this kind of error. This represents some long lasting exceptional system condition, which can luckily be modeled using Celix condition. Once some outside watchers get notified of this system exception, he/she can try to restart the bundle, which could fix the accidentally closed fd in our case.

I think generic system exception is a generally useful concept, which further extends error code and C++ exception. WDYT?

Reply:
Concerning the `sleep(1)` / retry on a error rc. I think this is fine, but currently the running is not updated in the else branch of `if (result > 0`) and this is an issue. 

I think the running update can be moved to the end part of the while loop:
```
while (running) {
 ....
  celixThreadMutex_lock(&announcer->mutex);
  running = announcer->running;
  celixThreadMutex_unlock(&announcer->mutex);
}
```

ensuring that running is updated every loop.

Reply:
> We can really do nothing locally to recover from this kind of error. This represents some long lasting exceptional system condition, which can luckily be modeled using Celix condition. Once some outside watchers get notified of this system exception, he/she can try to restart the bundle, which could fix the accidentally closed fd in this case.
> 
> I think generic system exception is a generally useful concept, which further extends error code and C++ exception. Combined with RSA, this means these exceptions can be watched remotely (cool). Event better, we don't need descriptors for Remote Celix conditions because they have no methods.
> 
> WDYT? @pnoltes @xuzhenbao

I think remote conditions are very useful and indeed this does not need a descriptor, because conditions are marker interfaces. remote conditions combined with a dependency management component means that components can react (start/stop) based on the presence of a condition in a different framework. 

Currently I think the RSA expect to find a dfi descriptor, so maybe an extensions is needed to service exported with descriptors are handled as marker interface services. 

btw of course out of scope for this PR.